AstSemantics.md

```diff
@@ -96,107 +120,59 @@ following rules:
-
-  * `load_mem_with_offset`: load a value from memory at a given index plus a
-    given immediate offset
-  * `store_mem_with_offset`: store a given value to memory at a given index
-    plus a given immediate offset
-
-The addition of the offset and index is specified to use infinite precision such
-that an out-of-bounds access never wraps around to an in-bounds access.  Bounds
-checking before the final offset addition allows the offset addition to easily
-be folded into the hardware load instruction *and* for groups of loads with the
-same base and different offsets to easily share a single bounds check.
+### Alignment
 
-In the MVP, the indices are 32-bit unsigned integers. With
-[64-bit integers](PostMVP.md#64-bit-integers) and
-[>4GiB memory](FutureFeatures.md#heaps-bigger-than-4gib), these nodes would also
-accept 64-bit unsigned integers.
+Each linear memory access operation also has an immediate positive integer power
+of 2 alignment attribute. An alignment value which is the same as the memory
+attribute size is considered to be a *natural* alignment.
 
-In the MVP, linear memory is not shared between threads. When
-[threads](PostMVP.md#threads) are added as a feature, the basic
-`load_mem`/`store_mem` nodes will have the most relaxed semantics specified in
-the memory model and new memory-access nodes will be added with atomic and
-ordering guarantees.
+If the effective address of a memory access is a multiple of the alignment
+attribute value of the memory access, the memory access is considered *aligned*,
+otherwise it is considered *misaligned*. Aligned and misaligned accesses have
+the same behavior. Alignment affects performance as follows:
 
-Two important semantic cases are **misaligned** and **out-of-bounds** accesses:
+ * Aligned accesses with at least natural alignment are fast.
+ * Aligned accesses with less than natural alignment may be somewhat slower
+   (think: implementation makes multiple accesses, either in software or
+    in hardware).
+ * Misaligned access of any kind may be *massively* slower
+   (think: implementation takes a signal and fixes things up)
 
-### Alignment
-
-If the incoming index argument to a memory access is misaligned with respect to
-the alignment operand of the memory access, the memory access must still work
-correctly (as if the alignment operand was `1`). However, on some platforms,
-such misaligned accesses may incur a *massive* performance penalty (due to trap
-handling). Thus, it is highly recommend that every WebAssembly producer provide
-accurate alignment. Note: on platforms without unaligned accesses,
-smaller-than-natural alignment may result in slower code generation (due to the
-whole access being broken into smaller aligned accesses).
+Thus, it is recommend that WebAssembly producers align frequently-used data
+to permit the use of natural alignment access, and use loads and stores with
+the grestest alignment values practical, while always avoiding misaligned
+accesses.
```

Comment:
Could you also mention that a misaligned access which is partly out-of-bounds traps?